src/molecules/mixingrules/MixingRuleBase.h

```diff
@@ -3,17 +3,57 @@
 
 #include <string>
 
-class XMLfileUnits;
+#include "utils/Logger.h"
+#include "utils/mardyn_assert.h"
+#include "utils/xmlfileUnits.h"
 
 class MixingRuleBase {
 public:
-	MixingRuleBase(){};
-	virtual ~MixingRuleBase(){};
+	MixingRuleBase(){}
+	virtual ~MixingRuleBase(){}
+
+	virtual void readXML(XMLfileUnits& xmlconfig) {
+		int cid1;
+		int cid2;
+		xmlconfig.getNodeValue("@cid1", cid1);
+		xmlconfig.getNodeValue("@cid2", cid2);
+
+		if (cid1 == cid2) {
+			Log::global_log->error() << "Mixing rules: cid1 and cid2 must not be the same but are both " << cid1 << std::endl;
+			mardyn_exit(1);
+		} else if ((cid1 <= 0) or (cid2 <= 0)) {
+			Log::global_log->error() << "Mixing rules: cid1 and cid2 must be greater than zero" << std::endl;
+			mardyn_exit(1);
+		// Symmetry for mixing rules is assumed
+		// cid1 must be less than cid2
+		} else if (cid1 > cid2) {
+			_cid1 = cid2 - 1;  // component id - 1 to convert to internal format starting with 0
+			_cid2 = cid1 - 1;
```

Comment:
Do you mean the `- 1`? The internal `cids` start at 0 because they are stored in a vector. I started to change the vector to a map to make it more intuitive and resolve the problem with the 0. However, this is quite complex...

Do you mean the assignment of `cid1` to `_cid2`? This is to ensure that `cid1` < `cid2`. This is needed since the same assumption is present in [Comp2Param.cpp](https://github.com/ls1mardyn/ls1-mardyn/blob/ced288067298a55d87a8d3213a70e8cad3aa925c/src/molecules/Comp2Param.cpp#L41). To prevent this, we could also store the `xi` and `eta` in both half spaces of the `_mixingrules` "matrix", i.e. `_mixingrules[cid2][cid1] = _mixingrules[cid1][cid2]`. But then the information is duplicated and not used...

Reply:
ah ok so looking at the whole if with its four cases we can say that the first two are input sanitation / error handling and the last two simply sort the `cid`s by magnitude and "convert" them by subtracting 1?

In that case, I'll make a new suggestion there.